oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -350,15 +352,29 @@ public void reindex() throws CommitFailedException, IOException {
             INDEXING_PHASE_LOGGER.info("[TASK:INDEXING:START] Starting indexing");
             Stopwatch indexerWatch = Stopwatch.createStarted();
             try {
-
                 if (flatFileStores.size() > 1) {
                     indexParallel(flatFileStores, indexer, progressReporter);
                 } else if (flatFileStores.size() == 1) {
                     FlatFileStore flatFileStore = flatFileStores.get(0);
+                    TopKSlowestPaths slowestTopKElements = new TopKSlowestPaths(TOP_SLOWEST_PATHS_TO_LOG);
```

Comment:
Why not accommodate the `IndexingProgressReporter` to provide this information?

Reply:
It seems to me that adapting `IndexingProgressReporter` to expose this information would require some refactoring. I'd rather keep this simple and orthogonal to existing mechanisms, at least until being more sure of the value of logging the slowest elements.